src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/subquery/rewriter/SubqueryAliasRewriter.java

```diff
@@ -0,0 +1,171 @@
+    private int aliasSuffix = 0;
+    private final static String DOT = ".";
+
+    @Override
+    public boolean visit(MySqlSelectQueryBlock query) {
+        SQLExprTableSource from = (SQLExprTableSource) query.getFrom();
+        String tableName = from.getExpr().toString().replaceAll(" ", "");
+
+        if (from.getAlias() != null) {
+            tableScope.push(new Table(tableName, from.getAlias()));
+        } else {
+            from.setAlias(createAlias(tableName));
+            tableScope.push(new Table(tableName, from.getAlias()));
+        }
+        return true;
+    }
+
+    @Override
+    public boolean visit(SQLIdentifierExpr expr) {
+        if (inSelect(expr) || inWhere(expr) || inSubquery(expr)) {
+            rewrite(tableScope.peek(), expr);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean visit(SQLAllColumnExpr expr) {
+        if (inSelect(expr) ) {
+            ((SQLSelectItem) expr.getParent()).setExpr(createIdentifierExpr(tableScope.peek()));
+        }
+        return true;
+    }
+
+    private boolean inSelect(SQLIdentifierExpr expr) {
+        return expr.getParent() instanceof SQLSelectItem;
+    }
+
+    private boolean inSelect(SQLAllColumnExpr expr) {
+        return expr.getParent() instanceof SQLSelectItem;
+    }
+
+    private boolean inWhere(SQLIdentifierExpr expr) {
+        return expr.getParent() instanceof SQLBinaryOpExpr && !isESTable((SQLBinaryOpExpr) expr.getParent());
+    }
+
+    /**
+     * The table name in elasticsaerch could be "index/type". Which represent as SQLBinaryOpExpr in AST.
+     */
+    private boolean isESTable(SQLBinaryOpExpr expr) {
+        return expr.getOperator() == SQLBinaryOperator.Divide && expr.getParent() instanceof SQLExprTableSource;
+    }
+
+    private boolean inSubquery(SQLIdentifierExpr expr) {
+        return expr.getParent() instanceof SQLInSubQueryExpr;
+    }
+
+    @Override
+    public void endVisit(MySqlSelectQueryBlock query) {
+        tableScope.pop();
+    }
+
+    private void rewrite(Table table, SQLIdentifierExpr expr) {
+        String tableAlias = table.getAlias();
+        String tableName = table.getName();
+
+        String exprName = expr.getName();
+        if (exprName.startsWith(tableName + DOT) || exprName.startsWith(tableAlias + DOT)) {
+            expr.setName(exprName.replace(tableName + DOT, tableAlias + DOT));
+        } else {
+            expr.setName(String.join(DOT, tableAlias, exprName));
+        }
+    }
+
+    private SQLIdentifierExpr createIdentifierExpr(Table table) {
+        String newIdentifierName = String.join(DOT, table.getAlias(), "*");
+        return new SQLIdentifierExpr(newIdentifierName);
+    }
+
+    private String createAlias(String alias) {
+        return String.format("%s_%d", alias, next());
+    }
+
+    private Integer next() {
+        return aliasSuffix++;
+    }
+
+    /**
+     * Table Bean.
+     */
+    private static class Table {
```

Comment:
minor, personal preference, feel free to ignore (although I recommend to remove empty line before first member):

The formatting of this class feels a bit unnatural (getters in the beginning, field after, javadocs over private fields). Consider.

```
   /**
     * Table Bean.
     */
    private static class Table {
        private String name;
        private String alias;

         public Table(String name, String alias) {
            this.name = name;
            this.alias = alias;
        }

         /** @return name of the table */
         public String getName() {
            return name;
        }

         /** @return alias of the table */
         public String getAlias() {
            return alias;
        }
    }
```